spec.md

```diff
@@ -523,13 +518,295 @@ schema being used.
         "bullets": [
           "40 concurrent connections"
         ]
+      },
+      "schemas": {
+        "service_instance": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+            }
+          },
+          "update": {
+            "parameters": {
+              "$schema": "http://json-schema.org/draft-04/schema#",
+              "type": "object",
+              "propeties": {
+                "some-other-account": {
+                  "description": "Other fake account needed for update.",
+                  "type": "string"
+                },
+                "billing": {
+                  "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+                }
+              }
+            }
+          }
+        },
+        "service_binding": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1/bindings#AccountParameters"
+            }
+          }
+        }
       }
     }]
   }]
 }
 ```
 
 
+## JSON Schemas
+
+The JSON Schemas endpoint is used to define reusable definitions of JSON Schema
+Objects referenced throughout the [Catalog](#catalog-management). It will be the
+responsibility of the Platform to examine the response from the catalog to
+determine if the Broker is using `$ref` JSON Schemas. The Broker can signal to
+the Platform that JSON Schema components are supported by using the `osb_v2:`
+scheme. Components are a way to allow Platforms to only fetch sub-schemas.
```

Comment:
We need some text around the lifecycle of this data. For example, let's say we have the following flow:
- platform get the catalog, which includes the schema and it has osb_v2:/// refs in there
- hours go by and the platform doesn't retrieve any sub-component schemas
- broker updates its catalog and removes the references sub-component schema - perhaps because the plan was removed
- platform then tries to retrieve the sub-component schema because someone tries to use that service/plan. They'll get a 404

While the user will eventually get some kind of error when they try to talk to the broker, but in the meantime the platform has the wrong info and is presenting it to the user. I don't think the platform can assume that any 404 for a schema means it should re-get the catalog, plus it may not want to update all its services/plans at that time.

So, I'm not sure what we can say about this yet so I'd like to brainstorm a bit, but I think we need to say something about the requirements we're placing on the broker w.r.t. how these endpoints need to be in-sync